### PR TITLE
Commerce errors

### PR DESCRIPTION
**Pull request by ashkan18**
# Problem
https://artsyproduct.atlassian.net/browse/PURCHASE-958

# Solution
Start act upon [newly added](https://github.com/artsy/exchange/pull/400) `error` events coming from Exchange.

Adding that complicated the existing `Aprb.Views.CommerceSlackView` so ended up separating that file to multiple _view_ files for different types of slack messages.

# Extra (or you could call it technical debt)
Added missing tests for order views. Still adding more but this should be good to review.

## Dependency Injection?!
For adding tests for view layer, we had to mock few Gravity calls so ended up following [this dependency injection approach](http://blog.plataformatec.com.br/2015/10/mocks-and-explicit-contracts/) so now in test mode we use `GravityMock` and in other cases we use `Gravity` for fetching data from Gravity.

# Selfie

image

### CODE REVIEW

File: config/test.exs [hunk 1]
```diff
@@ -6,7 +6,10 @@ config :maru, Aprb.Api.Root,
 config :aprb, Aprb.Repo,
   pool: Ecto.Adapters.SQL.Sandbox,
   adapter: Ecto.Adapters.Postgres,
-  username: System.get_env("DB_USER"),
-  password: System.get_env("DB_PASSWORD"),
+  # username: System.get_env("DB_USER"),
+  # password: System.get_env("DB_PASSWORD"),
```
Comment (lilyfromseattle) [inline config/test.exs]: Did you mean to leave these commented out lines in here?

Comment (ashkan18) [reply config/test.exs]: this is little annoying actually, working on fixing it now, but the only way i can run the test outside of Hokusai is to comment this out 😞 

alternative is to setup user/pass locally for my test database and have these default to that...

**Comment by sweir27** [comment]
@ashkan18 is your screenshot correct? Does the most recent version include the `Code` as well?

**Comment by ashkan18** [comment]
@sweir27 it was old, (and there was a bug) fixed it :)

thanks all, merging 🙌

**Pull request merged**